### ISSUE TITLE
unexpected '0' expecting space

### ISSUE THREAD

**Issue by jmatsushita** [hadolint/hadolint#142]
### Expected behavior

Better error message about mismatched quoting symbols (The first key in the example opens with a double quote and ends with a simple quote).

### Actual behavior

```
/dev/stdin:3:27 unexpected '0' expecting space, "\t", "ONBUILD", "FROM", "COPY", "RUN", "WORKDIR", "ENTRYPOINT", "VOLUME", "EXPOSE", "ENV", "ARG", "USER", "LABEL", "STOPSIGNAL", "CMD", "SHELL", "MAINTAINER", "ADD", "#", "HEALTHCHECK" or end of input
``

### Steps to reproduce the behavior

Haskell Dockerfile Linter v1.2.3-rc1-6-g7df5f1c

```bash
docker run --rm -i hadolint/hadolint < Dockerfile
```

**Dockerfile** (if relevant)

```Dockerfile
FROM docker:latest

LABEL \
  key="value' \
  key2="0"
```

**Comment by zemanlx** [comment]
@lorenzo would you mind to take care of this in `language-docker`?  AFAIK hadolint is showing whatever message it gets from parser at this stage.

**Comment by lorenzo** [comment]
The parser for label is completely wrong, I was planning to re-implement it. Not sure if doing that will actually improve the error messages, though.

**Comment by lorenzo** [comment]
This is in practical terms, a issue that is not possible to solve since this is actually valid

```dockerfile
LABEL \
  key="value' \
  key2="
```
I'm closing since improving the error message in this case is not something we can solve in a sane way.
